File: Backend/Infrastructure/Mapper/PersonMapper.cs
using Domain.IModel;
using Domain.Model;
using Domain.Visitor;
using Infrastructure.DataModel;

namespace Infrastructure.Mapper;

public class PersonMapper
{
    public static IPersonVisitor? MapToDataModel(IPerson? person)
    {
        if (person == null) return null;

        return new PersonDataModel
        {
            Id = person.Id,
            Name = person.Name,
            Age = person.Age,
        };
    }

    public static IPerson MapToDomainModel(IPersonVisitor dataModel)
    {
        return new Person(
            dataModel.Id,
            dataModel.Name,
            dataModel.Age
        );
    }
}

File: Frontend/src/app/person/mapper/person.mapper.ts
import { Person } from "../Models/person";
import { PersonViewModel } from "../Models/person-view-model";

export function toPersonViewModel(p : Person):PersonViewModel{
    return {
        id : p.id,
        name : p.name, 
        age : p.age
    }
}

export function toPerson(p : PersonViewModel):Person{
    return {
        id : p.id,
        name : p.name, 
        age : p.age
    }
}